src/app/components/connect-applications-wizard/connect-applications-wizard.directive.coffee

```diff
@@ -8,7 +8,67 @@ angular.module 'mnoEnterpriseAngular'
         areAppsReady: "="
       }
 
-      controller: () ->
+      controller: ($scope, MnoeAppInstances, $window) ->
+        $scope.activeApp = null
+
+        $scope.isLaunchHidden = (app) ->
+          app.status == 'terminating' ||
+          app.status == 'terminated' ||
+          $scope.isOauthConnectBtnShown(app) ||
+          $scope.isNewOfficeApp(app)
+      
+        $scope.isOauthConnectBtnShown = (instance) ->
+          instance.app_nid != 'office-365' &&
+          instance.stack == 'connector' &&
+          !instance.oauth_keys_valid
+
+        $scope.isNewOfficeApp = (instance) ->
+          instance.stack == 'connector' && instance.appNid == 'office-365' && (moment(instance.createdAt) > moment().subtract({minutes:5}))
+
+        $scope.launchApp = (app, event) ->
+          $scope.setAppActive(event, app.id)
+          if app.customInfoRequired
```

Comment:
I can't find where `app.customInfoRequired`, I think it is dead code. This test can be removed from here. We will need to remove it from the dock as well.